content/en/docs/Tracing/tempo-walkthrough.md

```diff
@@ -0,0 +1,206 @@
+
+```shell
+helm upgrade --install kyverno --namespace kyverno --create-namespace --wait \
+  --devel --repo https://kyverno.github.io/kyverno kyverno \
+  --values - <<EOF
+# kyverno controller
+extraArgs:
+  # enable tracing
+  - --enableTracing
+  # tempo backend url
+  - --tracingAddress=tempo.monitoring
+  # tempo backend port for opentelemetry traces
+  - --tracingPort=4317
+
+# cleanup controller
+cleanupController:
+  tracing:
+    # enable tracing
+    enabled: true
+    # tempo backend url
+    address: tempo.monitoring
+    # tempo backend port for opentelemetry traces
+    port: 4317
+EOF
+```
+
+## Kyverno policies Setup
+
+Finally we need to deploy a couple of policies in the cluster so that Kyverno can configure admission webhooks accordingly.
+
+We are going to deploy the `kyverno-policies` helm chart (with the `Baseline` profile of PSS) using the following command:
+
+```shell
+helm upgrade --install kyverno-policies --namespace kyverno --create-namespace --wait \
+  --devel --repo https://kyverno.github.io/kyverno kyverno-policies \
+  --values - <<EOF
+validationFailureAction: Enforce
+EOF
+```
+
+Note that we are setting `validationFailureAction` to `Enforce`, this is because `Audit` policies are processed asynchronously and will produce a separate trace from the main one (both traces are linked together though, but not with a parent/child relationship).
+
+## Create a Pod and observe the corresponding trace
+
+With everything in place we can exercise the Kyverno admission webhooks by creating a `Pod` and find the corresponding trace in Grafana.
+
+Run the following command to create a `Pod`:
+
+```shell
+kubectl run nginx --image=nginx
+```
+
+After that, navigate to the [Grafana explore page](http://localhost/grafana/explore), select `Tempo` in the top left drop down list, click on the `Search` tab, and search for traces with the following criterias:
+- Service name: `kyverno`, every trace define a service name and all traces coming from Kyverno will use the `kyverno` service name
+- Span name: `ADMISSION POST /validate/fail`, every span define a span name and root spans created by Kyverno when receiving an admission request have their name computed from the http operation and path (`ADMISSION <HTTP OPERATION> <HTTP PATH>`, the validate `/validate/fail` path indicates indicates that it's a validating webhook that was configured to fail the admission request in case of error).
+
+The list should show the trace for the previous `Pod` creation request:
+
+<p align="center"><img src="../assets/walkthrough-tempo-1.png" height="300px"/></p>
+
+Clicking on the trace will take you to the trace details, showing all spans covered by the `Pod` admission request:
+
+<p align="center"><img src="../assets/walkthrough-tempo-2.png" height="300px"/></p>
+
+The trace shows individual spans of all the policies that were just installed, with child spans for every rule that was checked (but not necessarily evaluated).
```

Comment:
One continual line here. Styling of Pod again.

Reply:
> One continual line here

Not sure, what shall I do ?

Reply:
Ah, I usually find it easier to read this way. I will remove the unnecessary line breaks.

Reply:
Fixed